eventlet/support/greendns.py

```diff
@@ -178,16 +187,16 @@ def _readlines(self):
 
         Note that this performs disk I/O so can be blocking.
         """
-        lines = []
         try:
-            with open(self.fname, 'rU') as fp:
-                for line in fp:
-                    line = line.strip()
-                    if line and line[0] != '#':
-                        lines.append(line)
+            with open(self.fname, 'rb') as fp:
+                fdata = fp.read()
         except (IOError, OSError):
-            pass
-        return lines
+            return []
+
+        if not isinstance(fdata, str):  # Python 3
```

Comment:
The `_readlines` function returns a list of `str`. On Python 2, `str` is the same as `bytes`. On Python 3 it is not. That is why I have to check the Python version.